File: app/models/course.rb
class Course < ActiveRecord::Base
  belongs_to :department, counter_cache: true
  has_many :sections
  #has_many :professors, -> { uniq }, through: :sections
  has_many :course_attributes
  has_many :cattributes, through: :course_attributes, class_name: "Attribute"

  validates :department, presence: true
end

File: app/assets/javascripts/models/course.js
// for more details see: http://emberjs.com/guides/models/defining-models/

App.Course = DS.Model.extend({
  title: DS.attr('string'),
  courseNum: DS.attr('string'),
  credits: DS.attr('number'),
  sectionsCount: DS.attr('number'),
  courseDescription: DS.attr('string'),
  department: DS.belongsTo('department'),
  cattributes: DS.hasMany('attribute', {async: true, key: 'cattribute_ids'}),
  sections: DS.hasMany('section', {async: true}),
  creditsValue: function() {
    var credits = this.get('credits');
    return credits >= 0 ? credits : 'V';
  }.property('credits'),
});
